Yosemite/YosemiteTests/Tools/CommonReaderConfigProviderTests.swift

```diff
@@ -0,0 +1,132 @@
+import Testing
+
+@testable import Yosemite
+import Networking
+
+struct CommonReaderConfigProviderTests {
+
+    @Test func fetchDefaultLocationID_returns_incompleteStoreAddress_error_when_jetpack_site_has_incomplete_store_address() async throws {
+        let adminURL = "https://example.com/wp-admin/set-address"
+        let mockRemote = MockCardReaderCapableRemote(
+            resultForDefaultReaderLocation: .failure(
+                DotcomError.unknown(code: "store_address_is_incomplete",
+                                    message: adminURL)))
+
+        let sut = CommonReaderConfigProvider(siteID: 123,
+                                             readerConfigRemote: mockRemote)
+
+        let expectedError = CardReaderConfigError.incompleteStoreAddress(adminUrl: URL(string: adminURL)!)
+        await #expect(throws: expectedError) {
+            try await withCheckedThrowingContinuation { continuation in
+                sut.fetchDefaultLocationID { result in
+                    continuation.resume(with: result)
+                }
+            }
+        }
+    }
+
+    @Test func fetchDefaultLocationID_returns_invalidPostalCode_error_when_jetpack_site_has_no_postcode() async throws {
```

Comment:
Nice. I haven't yet tried it!

Reply:
It was really easy to get started with, I can't see any reason that we shouldn't use it for new test files (and maybe just all new tests)